phper/src/classes.rs

```diff
@@ -559,10 +559,10 @@ impl<T: 'static> ClassEntity<T> {
     /// use phper::classes::{ClassEntity, ClassEntry};
     ///
     /// let mut class = ClassEntity::new("MyException");
-    /// class.extends(|| ClassEntry::from_globals("Exception").unwrap());
+    /// class.extends("Exception");
     /// ```
-    pub fn extends(&mut self, parent: impl Fn() -> &'static ClassEntry + 'static) {
-        self.parent = Some(Box::new(parent));
+    pub fn extends(&mut self, parent_name: impl Into<String>) {
```

Comment:
I think this would be better if it accepted a ClassEntry (which could come from a just-created class or defer looking up from globals, similar to `Interface::from_name`), but I couldn't make it work. I think this is at least more ergonomic than the previous way?

Reply:
How about `StateClass<()>`? Add method `from_name` just for `StateClass<()>`.

Reply:
The parameter can accept `StateClass<T>`, which can be cast to `StateClass<()>` internally. Anyway, `T` is just phantom data in `StateClass`.

Reply:
Thanks for the pointer. Some progress, now we can either `extends(StateClass<T>)` or `extends_name(String)`, the second one being useful for built-ins that we don't have a StateClass for. I _could_ allow lazily creating a StateClass for built-ins, to unify those two methods - WDYT?

Reply:
In fact, I prefer to change it this way to be consistent with the implements method. What do you think?

https://github.com/phper-framework/phper/compare/master...jmjoy:phper-fork:extends